conan/tools/meson/toolchain.py

```diff
@@ -3,13 +3,85 @@
 from jinja2 import Template
 
 from conan.tools._check_build_profile import check_using_build_profile
-from conan.tools.apple.apple import to_apple_arch
+from conan.tools.apple.apple import to_apple_arch, is_apple_os
 from conan.tools.cross_building import cross_building, get_cross_building_settings
 from conan.tools.env import VirtualBuildEnv
+from conan.tools.meson.helpers import *
 from conan.tools.microsoft import VCVars
+from conans.errors import ConanException
 from conans.util.files import save
 
 
+class _MesonAppleBlock(object):
+
+    def __init__(self, conanfile, toolchain):
+        self._conanfile = conanfile
+        self._toolchain = toolchain
+
+        os_ = conanfile.settings.get_safe("os")
+        if not is_apple_os(os_):
+            return
+        # SDK path is mandatory for cross-building
+        sdk_path = self._conanfile.conf["tools.meson.mesontoolchain:sdk_path"]
+        if not sdk_path and self._toolchain.cross_build:
+            raise ConanException("You must provide a valid SDK path for cross-compilation.")
+
+        os_version = self._conanfile.settings.get_safe("os.version")
+        os_subsystem = self._conanfile.settings.get_safe("os.subsystem")
+        # It's better if os.sdk field appears in settings instead of inferring it from SDK path
+        # os_sdk = os.path.basename(sdk_path).split(".")[0].lower()
+        os_sdk = _MesonAppleBlock.apple_sdk_name(self._conanfile)
+        arch = to_apple_arch(self._conanfile.settings.get_safe("arch"))
+        # Calculating main flags
+        deployment_target_flag = " " + _MesonAppleBlock.apple_min_version_flag(os_version, os_sdk,
+                                                                               os_subsystem)
+        sysroot_flag = " -isysroot " + sdk_path if sdk_path else ""
+        arch_flag = " -arch " + arch
+        # Adding all the flags needed (it does not matter if they are duplicated)
```

Comment:
It really doesn't matter if they are duplicated?

Reply:
That's it. I made a test putting the same flags twice and all the tests passed so I think it's good enough.

Reply:
Because you are putting exactly the same value. But what happens if they get different values? (a mismatch of what the user defines in profile, and the automatically added by Conan)

Reply:
Yes, you're right. Let me add a method to update properly these flags and a test to verify it.